### ISSUE TITLE
Remove android:requestLegacyExternalStorage 

### ISSUE THREAD

**Issue by cammellos** [status-im/status-react#12063]
# Problem

Because of some Policy changes on the play store https://developer.android.com/about/versions/11/privacy/storage 
we won't be able to use this permission anymore from the 5th of May.

One of the libraries we have is currently using it https://github.com/react-native-cameraroll/react-native-cameraroll/issues/304 and seems like it will take some time for them to upgrade to the new system.

My understanding that this will become a problem once we start targeting API 30 (we currently target API 29).

We will have to target API 30 by August (or November, still not very clear).

Here's a recap from a developer for the changes:

https://www.reddit.com/r/androiddev/comments/mwaqn1/scoped_storage_recap/

My understanding is that:

- From the 5th of May, if we target API 30, we will not be able to ask for android:requestLegacyExternalStorage and we would have instead to ask for `MANAGE_EXTERNAL_STORAGE`, but we'd have to ask google permission to allow our use case (unlikely to happen).

- If we don't target API 30 (our case), we can still use android:requestLegacyExternalStorage and publish updates to the play store, until we move to API 30. This will have to happen for August or November (to clarify).

**Comment by dokkis** [comment]
Hey, I have done some digging, researching and testing. I am still not 100% sure but I can start sharing my knowledge and understanding:

1. to me, it seems like they are not running a campaign to remove requestLegacyExternalStorage. requestLegacyExternalStorage can still be used. The campaign is about notifying that if you're using this flag, for devices running on Android 11 (if you target Android 11), this flag will be ignored. So in case you need broad access to files you should either use alternative APIs or request MANAGE_EXTERNAL_STORAGE permissions and obtain exception (as it's not privacy friendly).
2. From the email they sent and some doc, this is not clear, but the flag requestLegacyExternalStorage will still work on Android 10 even if you target Android 11. As stated here, https://developer.android.com/training/data-storage/use-cases:
`Caution: After you update your app to target Android 11 (API level 30), the system ignores the requestLegacyExternalStorage attribute when your app is running on Android 11 devices, so your app must be ready to support scoped storage and to migrate app data for users on those devices.`

3. with just point 1 and 2 seems that still some code changes are needed. There is another point. I tested the library on Android 11 targeting Android 11 and I noticed that the library still works, but couldn't understand why till I went in this documentation: https://developer.android.com/about/versions/11/privacy/storage#media-file-access
```
Access files using direct file paths and native libraries
To help your app work more smoothly with third-party media libraries, Android 11 allows you to use APIs other than the MediaStore API to access media files from shared storage using direct file paths. These APIs include the following:

The File API.
Native libraries, such as fopen().
```

To summarise, from the research and test I did we can keep the flag (so the app will not break on Android 10) and on Android 11 should still work because they implemented a friendly but still privacy compliant capability, where File APIs can still access media storage, and I am guessing that's the reason why I don't see the save functionality failing on Android 11.

Can somebody else test this as well and double check if that's the case?

Thanks,
Antonio
